### PR TITLE
[Solana]: Add utils to get/set compute unit price/limit

### PR DESCRIPTION
**Pull request by satoshiotomakan**
## Description

We want to be able to update `ComputeBudgetInstruction::SetComputeUnitPrice` and `ComputeBudgetInstruction::SetComputeUnitLimit` instructions for every transaction goes through Trust Wallet including Swap transactions.

Context: Rango (later Amber), Jupiter and most of the providers return a serialized transaction to sign and broadcast.

Solution: We should be able to deserialize the given raw transaction, add or update `ComputeBudgetInstruction::SetComputeUnitPrice` and `ComputeBudgetInstruction::SetComputeUnitLimit` instructions, compile the transaction again, and sign it.

## How to test

Run Rust, C++, iOS, Android tests

## Types of changes

* SolanaTransaction.getComputeUnitLimit(encodedTransaction: String) -> String?, where:
  * encodedTransaction - base64 encoded transaction
  * return - optional Unit Limit as a decimal string
* SolanaTransaction.setComputeUnitLimit(encodedTransaction: String, limit: String) -> String, where:
  * encodedTransaction - base64 encoded transaction
  * limit - decimal string
  * return - base64 encoded updated transaction
* SolanaTransaction.getComputeUnitPrice(encodedTransaction: String) -> String?, where
  * encodedTransaction - base64 encoded transaction
  * return - optional Unit Price as a decimal string
* SolanaTransaction.setComputeUnitPrice(encodedTransaction: String, price: String) -> String, where:
  * encodedTransaction - base64 encoded transaction
  * price - decimal string
  * return - base64 encoded updated transaction

## Checklist

- [x] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

**Comment by satoshiotomakan** [comment]
Hi @10gic, could you please review the PR as it's related to https://github.com/trustwallet/wallet-core/pull/4082

**Review by Milerius — APPROVED** [review]

**Review by JaimeToca — APPROVED** [review]

File: rust/chains/tw_solana/src/modules/utils.rs [hunk 1]
```diff
@@ -75,4 +82,113 @@ impl SolanaTransaction {
...
+        let ix_position = instructions
+            .iter()
+            .position(|ix| try_instruction_as_set_unit_limit(ix).is_some());
+        // If it presents already, it's enough to update the instruction data only.
+        if let Some(pos) = ix_position {
+            tx.message.instructions_mut()[pos].data = set_limit_ix.data;
+            return tx.to_base64().tw_err(|_| SigningErrorType::Error_internal);
+        }
+
+        // `ComputeBudgetInstruction::SetComputeUnitLimit` should be at the beginning of the instructions list.
```
Comment (10gic) [inline rust/chains/tw_solana/src/modules/utils.rs:145]: Hi @satoshiotomakan, I was curious why we should place the SetComputeUnitLimit at the beginning of the instructions list. Could you provide more detailed information?

Comment (satoshiotomakan) [reply rust/chains/tw_solana/src/modules/utils.rs:145]: Hi @10gic, I won't try to explain better than this article :)
https://www.helius.dev/blog/priority-fees-understanding-solanas-transaction-fee-mechanics
Jump into the `Best Practices` section

Comment (10gic) [reply rust/chains/tw_solana/src/modules/utils.rs:145]: Hi @satoshiotomakan, thanks for sharing this link. I suspect that this is outdated information. I found a transaction with 9 instructions (https://explorer.solana.com/tx/2Sca3UiTEaqQ7B9xPsL36gW8zRuBMRK5vDeUGG2Twe8zYfrVz1pwvAzbKHgRERsnDbVKQ6CUGCGJF1BiSQSafCCX), and the SetComputeUnitLimit is the last instruction.

In this transaction, the 7th Instruction consumed more than the default limit (200000):
```
 Program consumed: 243455 of 354569 compute units
```

Comment (satoshiotomakan) [reply rust/chains/tw_solana/src/modules/utils.rs:145]: Good to know, thank you for the hint! I think it's fine that we insert the Limit at the beginning, so no need to open a PR to change it

Comment (10gic) [reply rust/chains/tw_solana/src/modules/utils.rs:145]: Placing SetComputeUnitLimit at the front is also fine.

I noticed that we lack test cases to ensure that the AdvanceNonceAccount instruction is always at the beginning.

I would be happy to provide a PR to move SetComputeUnitLimit to the end of the instructions and update the existing test cases, so we wouldn't need to write new ones to ensure that AdvanceNonceAccount is always at the beginning.

What do you think?

**Pull request merged**